kong/client.go

```diff
@@ -238,7 +238,8 @@ func (c *Client) DoRAW(ctx context.Context, req *http.Request) (*http.Response,
 func (c *Client) Do(ctx context.Context, req *http.Request,
 	v interface{},
 ) (*Response, error) {
-	resp, err := c.DoRAW(ctx, req)
+	// TODO https://github.com/Kong/go-kong/issues/273 clear this lint ignore
+	resp, err := c.DoRAW(ctx, req) //nolint:bodyclose
```

Comment:
Might have as well just added a `defer`ed `resp.Close()` 🤷

Reply:
Okay, that _does_ break the tests _horribly_: https://github.com/Kong/go-kong/actions/runs/4079171550/jobs/7030257554

Somehow that response body is `nil` when the deferred close files? What the heck are we doing that causes that? I guess somehow the copy and decode destroys it? Anyway, back to the original plan.

Reply:
The problem here is that we have an API: [`Client.Do()`](https://github.com/Kong/go-kong/blob/b5e8261e4507d18ab51d25c740152ada25466963/kong/client.go#L238) where we return a wrapper around `http.Response` where we don't close the body but it's already read (through `json.NewDecoder(resp.Body).Decode(v)`).

We should explicitly add a comment to that method that the response can be used to read e.g. the status code but the body will be completely read **BUT** the caller should close it nonetheless. Otherwise we might leak that.

Created #276 to track this.

Reply:
`DoRAW()` seem to be fine because it explicitly states in the comment that 

```
the caller is responsible for closing the response body
```

and body is not manipulated in anyway.